File: tests/kits/go/kit/kit.go
package kit
import "fmt"
type Agent struct {
	id int
}

func (agent Agent) Initialize() {
	var id int
	// get agent ID
	fmt.Scanf("%d", &id)
	agent.id = id
	var rounds int
	fmt.Scanf("%d", &rounds)
}

func (agent Agent) Update() {
	var result int
	// get agent ID
	fmt.Scanf("%d", &result)
	var opponentMove string
	fmt.Scanf("%s", &opponentMove)
}

func (agent Agent) EndTurn() {
	fmt.Printf("D_FINISH\n");
}

File: oldtests/kits/php/kit.php
<?php
class Agent {
    public $id;
    function  __construct () {

    }

    function initialize() {
        fscanf(STDIN, "%d\n", $id);
        fscanf(STDIN, "%d\n", $max_rounds);
    }
    function update() {
        fscanf(STDIN, "%s\n", $result);
        fscanf(STDIN, "%s\n", $opponentmove);
    }
    function end_turn() {
        print "D_FINISH\n";
        flush();
    }
}